### PR TITLE
Feat/loadbalancer

### PR DESCRIPTION
**Pull request by tufranz**
The loadbalancer branch with the latest `destroy.yml` from main.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,8 +129,13 @@ For more documentation see the [README](roles/cert_bot/README.md)
...
 ansible-playbook -i inventory/ --user=<sudo-user> --become ./cert_bot.yaml
 ```
+This playbook has an additional variable `var_hosts`. Default ist `'~.*-vice-haproxy\\..*'`. Change this var to the host or group the playbook should run on.
+
 
+```bash
+ansible-playbook -i inventory/ --user=<sudo-user>  --extra-vars="var_hosts=loadbalancer"--become ./cert_bot.yaml
+```
 
 # TODO
```
Comment (mb-wali) [inline README.md:139]: i think you can get rid of this TODO.

File: roles/haproxy-loadbalancer/templates/haproxy.cfg.j2 [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+frontend  localhost
+    mode http
+    bind *:443 ssl crt {{ tls_pem_path }}
+
+    # redirect all HTTP traffic to HTTPS
+    acl is_https ssl_fc
+    http-request redirect scheme https unless is_https
+
+    # keycloak
+    acl keycloak hdr(host) -i keycloak.cyverse.at
```
Comment (mb-wali) [inline roles/haproxy-loadbalancer/templates/haproxy.cfg.j2:73]: can we do this also based of vars?
```suggestion
    acl keycloak hdr(host) -i {{keycloak_host}}
```

File: roles/haproxy-loadbalancer/templates/haproxy.cfg.j2 [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+backend userportal
+    #redirect scheme https if !{ ssl_fc }
+    balance roundrobin
+    option  log-health-checks
+    option  forwardfor
+    # TODO server sX
+{% for host in groups['k8s-workers'] %}
+    server   {{ hostvars[host].ansible_hostname }} {{ hostvars[host].ansible_fqdn }}:30392 check
+{% endfor %}
+    # server  s1 k8-w01.cyverse.at:30392 check
```
Comment (mb-wali) [inline roles/haproxy-loadbalancer/templates/haproxy.cfg.j2:118]: you remove these hosts

File: roles/haproxy-loadbalancer/templates/haproxy.cfg.j2 [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+#---------------------------------------------------------------------
+backend keycloak
+    balance roundrobin
+    option  log-health-checks
+    option  forwardfor
+    # TODO server sX
+{% for host in groups['k8s-workers'] %}
+    server   {{ hostvars[host].ansible_hostname }} {{ hostvars[host].ansible_fqdn }}:31360 check
+{% endfor %}
+    # server  s1 k8-w01.cyverse.at:31360 check # worker nodes hostname
```
Comment (mb-wali) [inline roles/haproxy-loadbalancer/templates/haproxy.cfg.j2:101]: remove these hosts

File: roles/haproxy-loadbalancer/templates/haproxy.cfg.j2 (hunk 2 above)
Comment (mb-wali) [reply roles/haproxy-loadbalancer/templates/haproxy.cfg.j2:73]: also for other backend.

File: roles/haproxy-loadbalancer/templates/haproxy.cfg.j2 (hunk 3 above)
Comment (tufranz) [reply roles/haproxy-loadbalancer/templates/haproxy.cfg.j2]: This was to remind myself to change the hostname to `s<INDEX>`.
is done.

**Review by mb-wali — APPROVED** [review]

**Pull request merged**